apis/python/src/tiledbsoma/io/registration/signatures.py

```diff
@@ -0,0 +1,305 @@
+import json
+from dataclasses import dataclass
+from typing import Dict, Optional, Tuple
+
+import anndata as ad
+import pandas as pd
+import pyarrow as pa
+from typing_extensions import Self
+
+import tiledbsoma
+import tiledbsoma.logging
+from tiledbsoma._arrow_types import df_to_arrow
+
+
+def _string_dict_from_arrow_schema(schema: pa.Schema) -> Dict[str, str]:
+    """
+    Converts an Arrow schema to a string/string dict, which is easier on the eyes,
+    easier to convert from/to JSON for distributed logging, and easier to do del-key on.
+    """
+
+    def stringify_type(t: pa.DataType) -> str:
+        retval = str(t)
+        # As noted in the Signature class, we pre-check logic from the ingestor.
+        # As detailed elsewhere, Arrow string and large_string must map to TileDB
+        # string, which is large-only. Thus string and large_string form an equivalence
+        # class. Similarly for Arrow binary and large_binary.
+        if retval == "large_string":
+            return "string"
+        if retval == "large_binary":
+            return "binary"
+        return retval
```

Comment:
Optional: This could be implemented with a little less repetition with a dictionary:

```python
_TYPE_MAP = {
  "large_string": "string",
  "large_binary": "binary",
}

def _stringify_type(t: pa.DataType) -> str:
  str_t = str(t)
  return _TYPE_MAP.get(str_t, str_t)
```

I would also recommend pulling the `stringify_type` function to the root level; since it doesn’t use anything from its closure it doesn’t really gain anything from being in one.